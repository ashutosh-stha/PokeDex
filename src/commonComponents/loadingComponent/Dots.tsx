import React, {useEffect, useRef} from 'react';
import {StyleSheet, View, Animated, Easing} from 'react-native';

const Dots: React.FC = () => {
  const dot1Opacity = useRef(new Animated.Value(1)).current;
  const dot2Opacity = useRef(new Animated.Value(1)).current;
  const dot3Opacity = useRef(new Animated.Value(1)).current;

  useEffect(() => {
    Animated.loop(
      Animated.stagger(300, [
        Animated.timing(dot1Opacity, {
          toValue: 0.3,
          duration: 600,
          easing: Easing.linear,
          useNativeDriver: true,
        }),
        Animated.timing(dot2Opacity, {
          toValue: 0.3,
          duration: 600,
          easing: Easing.linear,
          useNativeDriver: true,
        }),
        Animated.timing(dot3Opacity, {
          toValue: 0.3,
          duration: 600,
          easing: Easing.linear,
          useNativeDriver: true,
        }),
        Animated.timing(dot1Opacity, {
          toValue: 1,
          duration: 600,
          easing: Easing.linear,
          useNativeDriver: true,
        }),
        Animated.timing(dot2Opacity, {
          toValue: 1,
          duration: 600,
          easing: Easing.linear,
          useNativeDriver: true,
        }),
        Animated.timing(dot3Opacity, {
          toValue: 1,
          duration: 600,
          easing: Easing.linear,
          useNativeDriver: true,
        }),
      ]),
    ).start();
  }, [dot1Opacity, dot2Opacity, dot3Opacity]);

  return (
    <View style={styles.container}>
      <Animated.View
        style={[
          styles.dot,
          {
            opacity: dot1Opacity,
          },
        ]}
      />
      <Animated.View
        style={[
          styles.dot,
          {
            opacity: dot2Opacity,
          },
        ]}
      />
      <Animated.View
        style={[
          styles.dot,
          {
            opacity: dot3Opacity,
          },
        ]}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  dot: {
    width: 10,
    height: 10,
    borderRadius: 5,
    backgroundColor: '#787878',
    marginHorizontal: 2,
  },
});

export default Dots;
