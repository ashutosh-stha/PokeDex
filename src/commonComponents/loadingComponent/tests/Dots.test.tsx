import React from 'react';
import {render} from '@testing-library/react-native';
import Dots from '../Dots';

const wrapper = render(<Dots />);

describe('Dots', () => {
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
