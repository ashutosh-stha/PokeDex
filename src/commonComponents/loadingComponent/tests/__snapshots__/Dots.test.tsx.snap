// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dots should render correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#787878",
        "borderRadius": 5,
        "height": 10,
        "marginHorizontal": 2,
        "opacity": 1,
        "width": 10,
      }
    }
  />
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#787878",
        "borderRadius": 5,
        "height": 10,
        "marginHorizontal": 2,
        "opacity": 1,
        "width": 10,
      }
    }
  />
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#787878",
        "borderRadius": 5,
        "height": 10,
        "marginHorizontal": 2,
        "opacity": 1,
        "width": 10,
      }
    }
  />
</View>
`;
